GPflowOpt/scaling.py

```diff
@@ -0,0 +1,229 @@
+
+        # These assignments take care of initial re-scaling of model data (they trigger setter properties)
+        self.X = model.X.value
+        self.Y = model.Y.value
+
+    def __getattr__(self, item):
+        """
+        If an attribute is not found in this class, it is searched in the wrapped model
+        """
+        return self.wrapped.__getattribute__(item)
+
+    def __setattr__(self, key, value):
+        """
+        If setting :attr:`wrapped` attribute, point parent to this object (the datascaler)
+        """
+        if key is 'wrapped':
+            object.__setattr__(self, key, value)
+            value.__setattr__('_parent', self)
+            return
+
+        super(DataScaler, self).__setattr__(key, value)
+
+    def __eq__(self, other):
+        return self.wrapped == other
+
+    @property
+    def input_transform(self):
+        """
+        Get the current input transform
+        :return: :class:`.DataTransform` input transform object
+        """
+        return self._input_transform
+
+    @input_transform.setter
+    def input_transform(self, t):
+        """
+        Configure a new input transform. Data in the model is automatically updated with the new transform.
+        :param t: :class:`.DataTransform` object: the new input transform.
+        """
+        assert(isinstance(t, DataTransform))
+        X = self.X.value
+        self._input_transform.assign(t)
+        self.X = X
+
+    @property
+    def output_transform(self):
+        """
+        Get the current output transform
+        :return: :class:`.DataTransform` output transform object
+        """
+        return self._output_transform
+
+    @output_transform.setter
+    def output_transform(self, t):
+        """
+        Configure a new output transform. Data in the model is automatically updated with the new transform.
+        :param t: :class:`.DataTransform` object: the new output transform.
+        """
+        assert (isinstance(t, DataTransform))
+        Y = self.Y.value
+        self._output_transform.assign(t)
+        self.Y = Y
+
+    @property
+    def normalize_output(self):
+        """
+        :return: boolean, indicating if output is automatically scaled to zero mean and unit variance.
+        """
+        return self._normalize_Y
+
+    @normalize_output.setter
+    def normalize_output(self, flag):
+        """
+        Enable/disable automated output scaling. If switched off, the output transform becomes the identity transform.
+        If enabled, data will be automatically scaled to zero mean and unit variance. When the output normalization is
+        switched on or off, the data in the model is automatically adapted.
+        :param flag: boolean, turn output scaling on or off
+        """
+
+        self._normalize_Y = flag
+        if not flag:
+            self.output_transform = LinearTransform(np.ones(self.Y.value.shape[1]), np.zeros(self.Y.value.shape[1]))
+        else:
+            self.Y = self.Y.value
```

Comment:
I assume this trigger setattr which looks at the previously set self._normalize_Y

shouldnt output_transform be cleared or set to normalization?